app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuDataCollectionRobot.kt

```diff
@@ -0,0 +1,93 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+@file:Suppress("TooManyFunctions")
+
+package org.mozilla.fenix.ui.robots
+
+/**
+ * Implementation of Robot Pattern for the settings Data Collection sub menu.
+ */
+
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.Visibility
+import androidx.test.espresso.matcher.ViewMatchers.withContentDescription
+import androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withParent
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import org.hamcrest.CoreMatchers.allOf
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.assertIsEnabled
+import org.mozilla.fenix.helpers.click
+
+class SettingsSubMenuDataCollectionRobot {
+
+    fun verifyNavigationToolBarHeader() = assertNavigationToolBarHeader()
+
+    fun verifyDataCollectionOptions() = assertDataCollectionOptions()
+
+    fun verifyUsageAndTechnicalDataSwitchDefault() = assertUsageAndTechnicalDataSwitchDefault()
+
+    fun verifyMarketingDataSwitchDefault() = assertMarketingDataValueSwitchDefault()
+
+    fun verifyDataCollectionSubMenuItems() {
+        verifyDataCollectionOptions()
+        verifyUsageAndTechnicalDataSwitchDefault()
+        verifyMarketingDataSwitchDefault()
+    }
+
+    class Transition {
+        val mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())!!
+        fun goBack(interact: SettingsRobot.() -> Unit): SettingsRobot.Transition {
+            mDevice.waitForIdle()
+            goBackButton().click()
+
+            SettingsRobot().interact()
+            return SettingsRobot.Transition()
+        }
+    }
+}
+
+private fun goBackButton() =
+    onView(withContentDescription("Navigate up"))
+
+private fun assertNavigationToolBarHeader() = onView(
+    allOf(withParent(withId(R.id.navigationToolbar)),
+        withText("Data collection")))
+    .check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
+
+private fun assertDataCollectionOptions() {
+
+    onView(withText("Usage and technical data"))

```

Comment:
This is a general comment for all the String you are using for `onView(withText...)` and to prevent the error we saw with the string in different English locales. We should use `R.string.nameOfTheString` instead of the hard code string. Let me share an example: https://github.com/mozilla-mobile/fenix/blob/master/app/src/androidTest/java/org/mozilla/fenix/screenshots/MenuScreenShotTest.kt#L221

Reply:
yeah I get it. I will find out id from UI automator.

Reply:
![image](https://user-images.githubusercontent.com/63259972/79614262-ce18fc00-811d-11ea-826e-8c687769ed74.png)
So, the test is failing as there is "%S" in strings.xml .

Please find the related test failing screenshot

![image](https://user-images.githubusercontent.com/63259972/79614440-2e0fa280-811e-11ea-93c1-f0302af82f55.png)

Reply:
@isabelrios you can checkout here regarding you previous comment about marketing_data_description :)